boards: nxp: mimxrt1180_evk: fix LMA for CM7 from ITCM

DerekSnell opened the pull request:
11431a80 changed the node name of the FlexSPI node, breaking the LMA calculation.  This fixes the LMA to store the CM7 image at the correct address in flash.

Fixes #96251

lucien-nxp commented on boards/nxp/mimxrt1180_evk/Kconfig.defconfig:
```diff
@@ -33,10 +33,11 @@ DT_CHOSEN_IMAGE_M7 = nxp,m7-partition
 # Only adjust LMA if running from ITCM
 if !CM7_BOOT_FROM_FLASH
 	# Adjust the offset of the output image if building for RT18xx SOC ITCM
```
RT118X soc ITCM

decsny reviewed (dismissed):
the refactor is good too, thanks. the review comments are optional style nits.

Commit:
boards: nxp: mimxrt1180_evk: fix LMA for CM7 from ITCM

11431a80 changed the node name of the FlexSPI node, breaking the LMA
calculation.  This fixes the LMA to store the CM7 image at the correct
address in flash.

Signed-off-by: Derek Snell <derek.snell@nxp.com>

DerekSnell commented on boards/nxp/mimxrt1180_evk/Kconfig.defconfig:
```diff
@@ -33,10 +33,11 @@ DT_CHOSEN_IMAGE_M7 = nxp,m7-partition
+	FLEXSPI_BASE := $(dt_nodelabel_reg_addr_hex,flexspi,1)
+	ITCM_BASE := $(dt_nodelabel_reg_addr_hex,itcm,1)
+	IMAGE_M7_ADDR := $(dt_chosen_reg_addr_hex,$(DT_CHOSEN_IMAGE_M7))
 	config BUILD_OUTPUT_ADJUST_LMA
```
Thanks @decsny , you know I love to make you happy.  Your nits have been addressed
